### ISSUE TITLE
[Bug]: [PartitionKey] Milvus crashed if creating a collection with partition_key_field=""

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#24163]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230515-8b8f9f1e
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus-2.2.9.dev8
```

### Current Behavior

Milvus crashed when creating a collection with partition key field is empty

### Expected Behavior

check the partition key field and return a meaningful error msg

### Steps To Reproduce

```markdown
schema = cf.gen_collection_schema(fields=[pk_field, int64_field, string_field, vector_field],
                                          partition_key_field="", auto_id=True)
c_name = cf.gen_unique_str("par_key")
collection_w, _ = self.collection_wrap.init_collection(name=c_name, schema=schema,
                                                               num_partitions=200)
```

### Milvus Log

panic: runtime error: index out of range [-1]
goroutine 1212 [running]:
github.com/milvus-io/milvus/internal/proxy.(*createCollectionTask).validatePartitionKey(0x3bda3c0?)
	/go/src/github.com/milvus-io/milvus/internal/proxy/task.go:190 +0x3f6
github.com/milvus-io/milvus/internal/proxy.(*createCollectionTask).PreExecute(0xc00657c050, {0xc006493680?, 0x1?})
	/go/src/github.com/milvus-io/milvus/internal/proxy/task.go:246 +0x47d

**Comment by xige-16** [comment]
will be fixed by pr https://github.com/milvus-io/milvus/pull/24167

**Comment by yanliang567** [comment]
verified fix on 2.2.0-20230517-d4f7ebd1
